File: Best Time to buy and sell stock.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        nums = prices
        minprice = nums[0]
        profit = 0
        for i in range(1 , len(nums)):
            if nums[i] < minprice:
                minprice = nums[i]
            elif nums[i] - minprice > profit:
                profit = nums[i] - minprice
        return profit

File: Best Time to Buy and Sell Stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int minprice = prices[0];
        int profit = 0;
        int n = prices.length;
        for(int i = 1; i <n; i++ ){
            if(prices[i] < minprice){
                minprice = prices[i];
            }
            else if(prices[i]- minprice > profit){
                profit = prices[i] - minprice;

            }
        
        }
        return profit;
    }
}
